.github/workflows/main.yml

```diff
@@ -1,18 +1,49 @@
-
-# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
-# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
-
 name: Python package
 
-on: [push]
+on:
+  push:
+    branches:
+      - master
+      - 2.5
+      - 3.0
+  pull_request:
+  workflow_dispatch:
 
 jobs:
-  build:
+  format:
+    # We want all subsequent steps of CI to refer to the fresh version of code,
+    # so run this before anything else.
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:  # https://github.com/stefanzweifel/git-auto-commit-action#checkout-the-correct-branch
+          ref: ${{ github.head_ref }}
+      - uses: actions/setup-python@v2
+      - run: pip install black
+      - run: black --check .
+      - name: If needed, commit black changes to the pull request
```

Comment:
i think we decided not to do this, but do use commit hooks? do you still want to do it?

Reply:
Correct! Swapped this out with precommit and associated configs.